lib/src/annotation.dart

```diff
@@ -2,24 +2,22 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-library source_gen.annotation;
-
-import 'dart:io';
 import 'dart:mirrors';
 
 import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/element/element.dart';
 import 'package:analyzer/dart/element/type.dart';
 import 'package:analyzer/src/dart/element/element.dart';
 import 'package:analyzer/src/generated/constant.dart';
-import 'package:analyzer/src/generated/resolver.dart';
 import 'package:analyzer/src/generated/utilities_dart.dart';
-import 'package:path/path.dart' as p;
+
+import 'constants.dart';
+import 'type_checker.dart';
 
 dynamic instantiateAnnotation(ElementAnnotation annotation) {
   var annotationObject = annotation.constantValue;
   try {
-    return _getValue(annotationObject, annotation.element.context.typeProvider);
+    return _getValue(annotation.constantValue);
```

Comment:
Just ran a little test – in all cases `constantValue` and `computeConstantValue` return the same result. 🤷‍♂️